src/libaktualizr/primary/sotauptaneclient.cc

```diff
@@ -447,6 +442,15 @@ bool SotaUptaneClient::getNewTargets(std::vector<Uptane::Target> *new_targets, u
         is_new = true;
       }
 
+      if (primary_ecu_serial == ecu_serial) {
+        if (!target.IsOstree() &&
+            (config.pacman.type == PackageManager::kOstree || config.pacman.type == PackageManager::kOstreeDockerApp)) {
+          LOG_ERROR << "Cannot install a non-OSTree package on an OSTree system";
+          sendEvent<event::InstallTargetComplete>(primary_ecu_serial, false);
```

Comment:
In other cases when `is_new = false` the event `event::InstallTargetComplete` is not sent, I am just wondering if it's correct place to do it here.